### PR TITLE
Feature/automated events

### PR DESCRIPTION
**Pull request by BenjixD**
## Automated Events
- Created a new GM event engine
- Core event loop builds off of MapleEvent and a EventScheduler that can be tuned to run any new event which conforms to the tools provided in MapleEvent

## Maple OX Quiz
- 20 questions, 10 seconds each
- uses Open Trivia DB (https://opentdb.com/) as 3rd party API
- Timed event everyday at 10PM EST (7PM PST)

### CODE REVIEW

File: src/net/server/channel/Channel.java [hunk 1]
```diff
@@ -485,4 +477,8 @@ public boolean setDojoProgress(int dojoMapId) {
     public long getDojoFinishTime(int dojoMapId) {
         return dojoFinishTime[getDojoSlot(dojoMapId)];
     }
+
+    public MapleEvent getEvent() {
+        return event;
+    }
```
Comment (BenjixD) [inline src/net/server/channel/Channel.java:483]: I actually want to remove this - the legacy GM events still use this function but I plan on updating the other events to follow the new core structure.

File: src/server/events/gm/MapleCoconut.java [hunk 2]
```diff
@@ -45,12 +45,12 @@
...
        public MapleCoconut(MapleMap map) {
-           super(1, 50);
+           super(map, 50);
            this.map = map;
        }
 
        public void startEvent() {
-           map.startEvent();
+           map.startEvent(null);
            for (int i = 0; i < 506; i++) {
```
Comment (BenjixD) [inline src/server/events/gm/MapleCoconut.java:54]: breaking change for MapleCoconut, but this event probably didnt work to begin with.

File: src/server/events/gm/MapleOxQuiz/QuestionStep.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+			removeClock();
+			checkAnswer(entry.getValue());
+			Thread.sleep(8 * 1000);
+		}
+
+		// Show Ranking
+		showRanking();
+	}
+
+	private void getProblems(int numQuestions) {
```
Comment (nansonzheng) [inline src/server/events/gm/MapleOxQuiz/QuestionStep.java:73]: numQuestions is unused, how did you keep track of the number of questions asked?

**Review by nansonzheng — DISMISSED** [review]
Ok

**Review by nansonzheng — APPROVED** [review]
Ok

File: src/server/events/gm/MapleOxQuiz/QuestionStep.java (hunk 3 above)
Comment (BenjixD) [reply src/server/events/gm/MapleOxQuiz/QuestionStep.java:73]: good catch, using string.format passed in num questions

**Pull request merged**